cpp/lib/src/ZoteroHarvesterConversion.cc

```diff
@@ -514,43 +514,63 @@ void PostProcessAuthorName(std::string * const first_name, std::string * const l
 }
 
 
-void IdentifyMissingLanguage(MetadataRecord * const metadata_record, const Config::JournalParams &journal_params) {
-    const unsigned minimum_token_count(5);
+void DetectLanguage(MetadataRecord * const metadata_record, const Config::JournalParams &journal_params) {
+    // Normalize given language
+    if (not Config::IsAllowedLanguage(metadata_record->language_)) {
+        LOG_WARNING("Removing invalid language: " + metadata_record->language_);
+        metadata_record->language_ = "";
+    } else if (not Config::IsNormalizedLanguage(metadata_record->language_)) {
+        const std::string normalized_language(Config::GetNormalizedLanguage(metadata_record->language_));
+        LOG_DEBUG("Normalized language: " + metadata_record->language_ + " => " + normalized_language);
+        metadata_record->language_ = normalized_language;
+    }
 
+    // attempt to automatically detect the language
     if (journal_params.language_params_.expected_languages_.empty())
         return;
 
-    if (journal_params.language_params_.expected_languages_.size() == 1) {
-        metadata_record->language_ = *journal_params.language_params_.expected_languages_.begin();
-        LOG_DEBUG("language set to default language '" + metadata_record->language_ + "'");
-        return;
-    }
+    std::string detected_language;
+    if (journal_params.language_params_.expected_languages_.size() == 1)
+        detected_language = *journal_params.language_params_.expected_languages_.begin();
```

Comment:
I think that you can use front() instead of begin().  Then you don't need the asterisk.

Reply:
No you can't, because this is a std::set and not a std::vector.